stl/inc/flat_map

```diff
@@ -0,0 +1,873 @@
+        return *this;
+    }
+
+    // [container.reqmts] iterators
+    _NODISCARD iterator begin() noexcept {
+        return iterator{_Data.keys.cbegin(), _Data.values.begin()};
+    }
+
+    _NODISCARD const_iterator begin() const noexcept {
+        return const_iterator{_Data.keys.cbegin(), _Data.values.begin()};
+    }
+
+    _NODISCARD iterator end() noexcept {
+        return iterator{_Data.keys.cend(), _Data.values.end()};
+    }
+
+    _NODISCARD const_iterator end() const noexcept {
+        return const_iterator{_Data.keys.cend(), _Data.values.end()};
+    }
+
+    _NODISCARD reverse_iterator rbegin() noexcept {
+        return _STD make_reverse_iterator(end());
+    }
+
+    _NODISCARD const_reverse_iterator rbegin() const noexcept {
+        return _STD make_reverse_iterator(end());
+    }
+
+    _NODISCARD reverse_iterator rend() noexcept {
+        return _STD make_reverse_iterator(begin());
+    }
+
+    _NODISCARD const_reverse_iterator rend() const noexcept {
+        return _STD make_reverse_iterator(begin());
+    }
+
+    _NODISCARD const_iterator cbegin() const noexcept {
+        return const_iterator{_Data.keys.cbegin(), _Data.values.cbegin()};
+    }
+
+    _NODISCARD const_iterator cend() const noexcept {
+        return const_iterator{_Data.keys.cend(), _Data.values.cend()};
+    }
+
+    _NODISCARD const_reverse_iterator crbegin() const noexcept {
+        return _STD make_reverse_iterator(cend());
+    }
+
+    _NODISCARD const_reverse_iterator crend() const noexcept {
+        return _STD make_reverse_iterator(cbegin());
+    }
+
+    void swap(flat_map& _Other) noexcept {
+        _STD swap(_Key_compare, _Other._Key_compare);
+        _Data.keys.swap(_Other._Data.keys);
+        _Data.values.swap(_Other._Data.values);
+    }
+
+    // [container.reqmts] clear
+    void clear() noexcept {
+        _Data.keys.clear();
+        _Data.values.clear();
+    }
+
+    // [flat.map.capacity] Capacity
+    _NODISCARD_EMPTY_MEMBER bool empty() const noexcept {
+        return _Data.keys.empty();
+    }
+
+    _NODISCARD size_type size() const noexcept {
+        return _Data.keys.size();
+    }
+
+    _NODISCARD size_type max_size() const noexcept {
+        return _STD min(_Data.keys.max_size(), _Data.values.max_size());
+    }
+
+    // [flat.map.access] Access
+    template <class _K>
+    _NODISCARD mapped_type& operator[](_K&& _Key)
+        requires same_as<remove_cvref_t<_K>, key_type>
+              || (_Compare_is_transparent<key_compare> && constructible_from<key_type, _K &&>)
+    {
+        return try_emplace(_STD forward<_K>(_Key)).first->second;
+    }
+
+    template <class _K>
+    _NODISCARD mapped_type& at(const _K& _Key)
+        requires same_as<_K, key_type> || _Compare_is_transparent<key_compare>
+    {
+        iterator _Position = find(_Key);
+        if (_Position == end()) {
+            throw _STD out_of_range("std::flat_map::at: the specified key does not exist.");

```

Comment:
```suggestion
            _Xout_of_range("std::flat_map::at: the specified key does not exist.");
```
Or use `_THROW(...)` instead of plain `throw`.

https://github.com/microsoft/STL/blob/6c69a73911b33892919ec628c0ea5bbf0caf8a6a/stl/src/xthrow.cpp#L24-L26